app/models/stripe_record.rb

```diff
@@ -36,28 +47,43 @@ class StripeRecord < ApplicationRecord
   # Also saves us from some pains because JSON columns are highly inconsistent among MySQL and MariaDB.
   serialize :parameters, coder: JSON
 
+  def determine_wca_status
+    WCA_TO_STRIPE_STATUS_MAP.each do |key, values|
+      return key if values.include?(stripe_status)
+    end
+
+    raise Exception("No associated wca_status for stripe_status: #{stripe_status} - our tests should prevent this from happening!")
```

Comment:
Excellent point. That presents a whole world of complexity. The best option I can think of is, if this point is reached, store an `unknown` status on the PaymentIntent - which at some point we may hopefully pick up? 

(We could even add a note on the frontend - perhaps in the `/registrations/edit` view - to alert WST because of a status inconsistency? Because otherwise this could go undetected for months, if `unknown` doesn't cause any errors to be thrown anywhere else in the code)